src/mahimahi.rs

```diff
@@ -99,3 +103,219 @@ pub trait MahimahiExt: Mahimahi {
+            }
+            std::cmp::Ordering::Equal => {
+                last_cnt += 1;
+            }
+            std::cmp::Ordering::Greater => {
+                if last_ts > 0 {
+                    // insert new bandwidth config
+                    insert_into_pattern(
+                        StaticBwConfig::new()
+                            .bw(MTU_PER_MILLIS * last_cnt)
+                            .duration(MAHIMAHI_TS_BIN),
+                    );
+                }
+                if ts - last_ts > 1 {
+                    // insert zero bandwidth
+                    insert_into_pattern(
+                        StaticBwConfig::new()
+                            .bw(Bandwidth::ZERO)
+                            .duration(MAHIMAHI_TS_BIN * ((ts - last_ts - 1) as u32)),
+                    );
+                }
+                last_cnt = 1;
+                last_ts = ts;
+            }
+        }
+    }
+    if last_cnt == 0 {
+        // no non-zero timestamps
+        return Err("trace must last for a nonzero amount of time");
+    } else {
+        // merge final timestamps and zero timestamps
+        insert_into_pattern(
+            StaticBwConfig::new()
+                .bw(MTU_PER_MILLIS * (last_cnt + zeor_ts_cnt))
+                .duration(MAHIMAHI_TS_BIN),
+        );
+    }
+    Ok(RepeatedBwPatternConfig::new().pattern(
+        pattern
+            .drain(..)
+            .map(|config| Box::new(config) as Box<dyn BwTraceConfig>)
+            .collect(),
+    ))
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::model::StaticBwConfig;
+    use crate::Bandwidth;
+
+    #[test]
+    fn test_trait() {
+        let mut static_bw = StaticBwConfig::new()
+            .bw(Bandwidth::from_mbps(24))
+            .duration(Duration::from_secs(1))
+            .build();
+        assert_eq!(
+            static_bw.mahimahi(&Duration::from_millis(5)),
+            [1, 1, 2, 2, 3, 3, 4, 4, 5, 5]
+        );
+        let mut static_bw = StaticBwConfig::new()
+            .bw(Bandwidth::from_mbps(12))
+            .duration(Duration::from_secs(1))
+            .build();
+        assert_eq!(
+            static_bw.mahimahi_to_string(&Duration::from_millis(5)),
+            "1\n2\n3\n4\n5"
+        );
+        let a = vec![
+            Box::new(
+                StaticBwConfig::new()
+                    .bw(Bandwidth::from_mbps(12))
+                    .duration(Duration::from_millis(2)),
+            ) as Box<dyn BwTraceConfig>,
+            Box::new(
+                StaticBwConfig::new()
+                    .bw(Bandwidth::from_mbps(24))
+                    .duration(Duration::from_millis(2)),
+            ) as Box<dyn BwTraceConfig>,
+        ];
+        let mut c = Box::new(RepeatedBwPatternConfig::new().pattern(a).count(2)).into_model();
+        assert_eq!(
+            c.mahimahi(&Duration::MAX),
+            [1, 2, 3, 3, 4, 4, 5, 6, 7, 7, 8, 8]
+        );
+    }
+
+    #[test]
+    fn test_load() {
```

Comment:
Test one more case that we want to load and play two mahimahi traces. For example, we can emulate a bandwidth trace where the first half is 24Mbps and the second half is 12Mbps by combining `[1,1,2,2,3,3]` and `[1,2,3]`.